File: app/src/main/java/com/example/mlallemant/mentalbattle/UI/Friends/FriendSearchAdapter.java
package com.example.mlallemant.mentalbattle.UI.Friends;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.example.mlallemant.mentalbattle.R;

import java.util.ArrayList;

/**
 * Created by m.lallemant on 13/11/2017.
 */

public class FriendSearchAdapter extends ArrayAdapter<FriendSearchModel> implements View.OnClickListener {

    private final ArrayList<FriendSearchModel> dataSet;
    Context mContext;

    private static class ViewHolder {
        TextView username;
    }

    public FriendSearchAdapter(final ArrayList<FriendSearchModel> data, final Context context) {
        super(context, R.layout.friend_search_row_item_template, data);
        this.dataSet = data;
        this.mContext = context;
    }

    @Override
    public void onClick(final View v) {
        final int position = (Integer) v.getTag();
        final Object object = getItem(position);

        final FriendSearchModel friendSearchModel = (FriendSearchModel) object;
    }

    private int lastPosition = -1;

    @Override
    public View getView(final int position, View convertView, final ViewGroup parent) {
        // Get the data item for this position
        final FriendSearchModel friendSearchModel = getItem(position);

        //Check if an existing view is being reused, otherwise inflate the view
        final ViewHolder viewHolder;

        final View result;

        if (convertView == null) {
            viewHolder = new ViewHolder();
            final LayoutInflater inflater = LayoutInflater.from(getContext());
            convertView = inflater.inflate(R.layout.friend_search_row_item_template, parent, false);

            viewHolder.username = (TextView) convertView.findViewById(R.id.friend_search_tv_username);

            result = convertView;
            convertView.setTag(viewHolder);
        } else {
            viewHolder = (ViewHolder) convertView.getTag();
            result = convertView;
        }

        lastPosition = position;

        if (friendSearchModel != null) {
            viewHolder.username.setText(friendSearchModel.player.getName());

        }
        return convertView;
    }
}

File: app/src/main/java/com/example/mlallemant/mentalbattle/ui/friends/FriendSearchAdapter.kt
package com.example.mlallemant.mentalbattle.ui.friends

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView
import com.example.mlallemant.mentalbattle.R
import java.util.*

/**
 * Created by m.lallemant on 13/11/2017.
 */
class FriendSearchAdapter(
    dataSet: ArrayList<FriendSearchModel?>,
    mContext: Context
) : ArrayAdapter<FriendSearchModel?>(
    mContext, R.layout.friend_search_row_item_template, dataSet
), View.OnClickListener {

    private class ViewHolder {
        var username: TextView? = null
    }

    override fun onClick(v: View) {
        val position = v.tag as Int
        val `object`: Any? = getItem(position)
        val friendSearchModel = `object` as FriendSearchModel?
    }

    private var lastPosition = -1

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        // Get the data item for this position
        val convertedView: View
        val friendSearchModel = getItem(position)

        //Check if an existing view is being reused, otherwise inflate the view
        val viewHolder: ViewHolder
        if (convertView == null) {
            viewHolder = ViewHolder()
            val inflater = LayoutInflater.from(context)
            convertedView = inflater.inflate(R.layout.friend_search_row_item_template, parent, false)
            viewHolder.username = convertedView.findViewById<View>(R.id.friend_search_tv_username) as TextView
            convertedView.tag = viewHolder
        } else {
            convertedView = convertView
            viewHolder = convertedView.tag as ViewHolder
        }
        lastPosition = position
        if (friendSearchModel != null) {
            viewHolder.username!!.text = friendSearchModel.player.name
        }
        return convertedView
    }
}